test/NuGet.Core.Tests/NuGet.Packaging.Test/ReferencesInNuspecMatchRefAssetsRuleTests.cs

```diff
@@ -0,0 +1,408 @@
+            };
+            var files = new string[]
+            {
+                "ref/MyLib.dll",
+                "ref/MyHelpers.dll"
+            };
+
+            //Act
+            var rule = new ReferencesInNuspecMatchRefAssetsRule();
+            var missingReferences = rule.Compare(references, files);
+
+            //Assert
+            Assert.Equal(missingReferences.Count(), 1);
+            var singleIssue = missingReferences.Single(t => t.MissingFrom.Equals("ref"));
+            Assert.Equal("net462", missingReferences.First().Tfm);
+            Assert.Equal(2, missingReferences.First().MissingItems.Count());
+            Assert.True(missingReferences.First().MissingItems.Contains("MyLib.dll"));
+            Assert.True(missingReferences.First().MissingItems.Contains("MyHelpers.dll"));
+        }
+
+        [Fact]
+        public void Compare_PackageWithAssetsUnderMiscellaneousFolders_ShouldBeEmpty()
+        {
+            //Arrange
+            var references = new Dictionary<string, IEnumerable<string>>()
+            {
+                { "MyCompany", new List<string>() { "MyLib.dll" } }
+            };
+            var files = new string[]
+            {
+                "ref/MyCompany/MyLib.dll",
+                "ref/MyCompany/MyHelpers.dll"
+            };
+
+            //Act
+            var rule = new ReferencesInNuspecMatchRefAssetsRule();
+            var missingReferences = rule.Compare(references, files);
+
+            //Assert
+            Assert.Empty(missingReferences);
+        }
+
+        [Fact]
+        public void Compare_PackageWithMismatchedReferencesInMultipleSubFolders_ShouldHaveCorrectItems()
+        {
+            //Arrange
+            var references = new Dictionary<string, IEnumerable<string>>
+            {
+                { "net462", new List<string>() { "MyLib.dll", "MyHelpers.dll" } },
+                { "net472", new List<string>() { "MyLib.dll" } }
+            };
+            var files = new string[]
+            {
+                "ref/net462/MyLib.dll",
+                "ref/net472/MyLib.dll",
+                "ref/net472/MyHelpers.dll"
+            };
+
+            //Act
+            var rule = new ReferencesInNuspecMatchRefAssetsRule();
+            var missingReferences = rule.Compare(references, files);
+
+            //Assert
+            Assert.Equal(missingReferences.Count(), 2);
+            var singleRefFilesMissing = missingReferences.Single(t => t.MissingFrom.Equals("ref"));
+            var singleReferencesMissing = missingReferences.Single(t => t.MissingFrom.Equals("nuspec"));
+            Assert.Equal("net462", singleRefFilesMissing.Tfm);
+            Assert.Equal(1, singleRefFilesMissing.MissingItems.Count());
+            Assert.Equal("net472", singleReferencesMissing.Tfm);
+            Assert.Equal(1, singleReferencesMissing.MissingItems.Count());
+            Assert.Equal("MyHelpers.dll", singleReferencesMissing.MissingItems[0]);
```

Comment:
just FYI, when writing tests consider if your test data is easy or difficult to understand. Here, the data is "realistic" in the sense that the assembly names across TFMs match. However, that makes it difficult when reading the tests to understand which file in which tfm is good or bad. If the references had files "one.dll", "two.dll" for one tfm and "three.dll", while the ref files had "two.dll" for the first tfm and "three.dll" and "four.dll", it would be so much easier to read the tests and be confident that the test itself doesn't have a bug.